docs/practices/pull_requests.md

```diff
@@ -0,0 +1,84 @@
+  Default to moving forward if you believe it is the right time to merge something.
+- **Use your best judgment**.
+  We do not provide many strict rules and technical barriers because we trust team members to use their best judgment in reviewing and merging work.
+  Do your best to live up to our team values and practices, and recognize that we all make mistakes.
+
+## When to merge pull requests
+
+- **Obvious and simple improvements** can be merged by any member of the {team}`Maintainer Team`, even if it does not have a review.
+  For example: clear bug fixes, typo corrections, etc.
+  If it's straightforward, go ahead and merge it.
+- **Straightforward and non-controversial PRs** with at least one approval from a {team}`Maintainer Team` member and no objections may be merged by anybody, anytime.
+  If you approve the PR, go ahead and merge it!
+  If you don't immediately merge it, provide a rationale for why not.
+  If somebody else approves your PR but doesn't merge and doesn't say why they didn't, feel free to self-merge.
+- **Complex or potentially controversial PRs** may be merged with at least one approval from a {team}`Maintainer Team` member, no objections from any team member, and after has been open for around five working days.
+  Use your best judgment for how long to wait to ensure inclusive decision making process and be sure to [include others in discussion](prs:principles).
+  The more complex, the more time you should give people.
+  The more people agree, the more quickly you can safely merge.
+- **Changes to team policies** may be merged with **at least two approvals** from {team}`JupyterHub Council` members, no objections from any team member, and has been open for around five working days (follow the same guidelines as above).
+- **Proposals without consensus** may be merged if the {team}`JupyterHub Council` has majority agreement to merge, and if there have been at least five working days to discuss.
+
+## What to look for in a review
+
+As a team member, you're encouraged to help others contribute to the project
+by reviewing their code, guiding them towards making a contribution and
+improving it, and ultimately merging their contribution into the project.
+
+Having merge rights is both a privilege and a responsibility - please be
+thoughtful when using it! To that extent, here are a few guidelines when
+deciding to merge things into one of our repositories[^pc].
+
+[^pc]: These are inspired by [Simon Willion's blog post on "The Perfect Commit"](https://simonwillison.net/2022/Oct/29/the-perfect-commit/).
+
+- **It's quality code**.
+  We know this is somewhat subjective, but ensure that the code is well-organized and thoughtfully-written, that any new features are documented, and that it abides by best-practices in Python, JavaScript, etc.
+  Do not nit-pick stylistic preferences, but if you genuinely think there's a clear way to improve the code, please make suggestions.
```

Comment:
I like this and wonder if we should expand it or add another line to say "don't nitpick" or "mark your nits as nits". I think a positive way to phrase it is "Try and stay big picture, worry about details later." Kinda "is this a change we want to make?", "is this a good way of doing it?", and then at the very end "should this loop variable be `n` or `i`?". I think I (and many others) are easily sucked into focusing on the details line-by-line of a PR instead of taking the step back. And often while you are looking at something you realise that "adjacent code" is not nice as well and could do with some tweaks and then ask for them. Instead I think we should either make an issue, or clearly mark such comments as "something for a new PR". All to try and not broaden the scope of a PR.

Reply:
I like the way that [consent-based decision-making often describes it](https://www.sociocracyforall.org/consent-decision-making/).

Basically, there is "what you want", then there is "what you will tolerate", and then there is "what you will not tolerate". We want people to approve PRs within "what they will tolerate", and we want people only to block merging if something has a major negative strategic implication on the project (after one or two rounds of expected minor improvements).

Reply:
See my comment about "merge because this is not worse than what we have". So I agree with your "merge what you can tolerate".

What I was trying to get at with my comment was that you should try and start your review with "big picture" thinking. And then later go line by line. And even later go character by character. Or maybe decide somewhere along the line, before reaching the end of this "zooming in", that this is super good enough and merge it.

Reply:
Marking this as resolved because of https://github.com/jupyterhub/team-compass/commit/3064410a420a6ba8e38ec372c5988a622390c206